pydantic_numpy/model.py

```diff
@@ -113,7 +112,19 @@ def load(
         field_to_value = {**npz_file, **other_field_to_value}
         if pre_load_modifier:
             field_to_value = pre_load_modifier(field_to_value)
-
+        
+        def extract_dirs_to_dict(dir_dict: dict[str, Any]) -> dict[str, Any]:
+            """ Extract from a flat dict of {a/b/c : value} to {a: {b: {c: value}}}"""
```

Comment:
You need to explain your nested dump and load strategy in the code somewhere so that newcomers can understand how different components work together to provide the function.

Use doc strings, and not comments.